DigitalLearningSolutions.Data.Tests/Services/PostLearningAssessmentTests.cs

```diff
@@ -419,6 +440,389 @@ public void Get_post_learning_assessment_next_section_skips_sections_full_of_arc
+                var result = postLearningAssessmentService.GetPostLearningAssessment(customisationId, candidateId, sectionId);
+
+                // Then
+                result.Should().NotBeNull();
+                result!.OtherSectionsExist.Should().BeTrue();
+            }
+        }
+
+        [Test]
+        public void Get_post_learning_assessment_should_not_have_otherSectionsExist_when_only_section_in_application()
+        {
+            // Given
+            const int customisationId = 7967;
+            const int candidateId = 11;
+            const int sectionId = 210;
+
+            // When
+            var result = postLearningAssessmentService.GetPostLearningAssessment(customisationId, candidateId, sectionId);
+
+            // Then
+            result.Should().NotBeNull();
+            result!.OtherSectionsExist.Should().BeFalse();
+        }
+
+        [Test]
+        public void Get_post_learning_assessment_should_not_have_otherSectionsExist_when_only_section_not_archived_in_course()
+        {
+            using (new TransactionScope())
+            {
+                // Given
+                const int customisationId = 21727;
+                const int candidateId = 210934;
+                const int sectionId = 1806;
+
+                // Make this course assessed
+                courseContentTestHelper.UpdateIsAssessed(customisationId, true);
+
+                // When
+                var result = postLearningAssessmentService.GetPostLearningAssessment(customisationId, candidateId, sectionId);
+
+                // Then
+                result.Should().NotBeNull();
+                result!.OtherSectionsExist.Should().BeFalse();
+            }
+        }
+
+        [Test]
+        public void Get_post_learning_assessment_should_not_have_otherSectionsExist_when_other_section_is_full_of_archived_tutorials()
+        {
+            using (new TransactionScope())
+            {
+                // Given
+                const int customisationId = 17456;
+                const int candidateId = 210934;
+                const int sectionId = 664;
+
+                // The tutorials of what would be the next section, 668
+                // This is the only other section on this course
+                tutorialContentTestHelper.ArchiveTutorial(2713);
+                tutorialContentTestHelper.ArchiveTutorial(2714);
+                tutorialContentTestHelper.ArchiveTutorial(2715);
+                tutorialContentTestHelper.ArchiveTutorial(2716);
+
+                // When
+                var result = postLearningAssessmentService.GetPostLearningAssessment(customisationId, candidateId, sectionId);
+
+                // Then
+                result.Should().NotBeNull();
+                result!.OtherSectionsExist.Should().BeFalse();
+            }
+        }
+
+        [Test]
+        public void Get_post_learning_assessment_should_have_otherSectionsExist_when_other_section_only_has_diagnostic_assessment()

```

Comment:
I think it would be useful to group all the "should be true" cases together and to group all the "should be false" cases together, since there are quite a few different cases 🙂

Reply:
Yeah I agree, done in 71a2952 🙂